### PR TITLE
AXI Wrapper for memory cmd/resp review

### PR DESCRIPTION
**Pull request by tanglingshu**
1) Adding bp_me_cce_mem_to_axi_wrapper.v for memory cmd/resp. Located under bp_me/src/v/axi/
2) Adding a testbench for mem_to_axi_wrapper. Located under bp_me/test/tb/bp_axi
3) Adding a draft version of bp_me_outgoing_io_to_axi_wrapper.v for outgoing IO cmd/resp.
4) Somehow the bp_top/syn/flist.vcs and the bp_top/test/tb/bp_tethered/testbench.v show differences. Not exactly sure what happened there.

### CODE REVIEW

**Comment by tanglingshu** [comment]
Seems like functions have changed in top_dev branch. For example, `declare_bp_me_if_widths is now `declare_bp_**mem**_if_widths. Will change first and make another PR

**Comment by dpetrisko** [comment]
You can keep this PR, just push and it’ll automatically update

File: bp_me/src/v/axi/bp_me_cce_mem_to_axi_wrapper.v [hunk 1]
```diff
@@ -0,0 +1,391 @@
+module bp_me_cce_mem_to_axi_wrapper
```
Comment (dpetrisko) [inline bp_me/src/v/axi/bp_me_cce_mem_to_axi_wrapper.v:1]: bp_mem_to_axi_master

File: bp_me/src/v/axi/bp_me_outgoing_io_to_axi_wrapper.v [hunk 2]
```diff
@@ -0,0 +1,393 @@
+module bp_me_outgoing_io_to_axi_wrapper
```
Comment (dpetrisko) [inline bp_me/src/v/axi/bp_me_outgoing_io_to_axi_wrapper.v:1]: bp_mem_to_axi_client

File: bp_me/src/v/axi/bp_me_cce_mem_to_axi_wrapper.v [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+  ( 
+  //==================== GLOBAL SIGNALS =======================
+    input aclk_i  
+  , input aresetn_i
+
+  //========================= BP side =========================
+  // memory commands from the ME into the axi wrapper
+  , input  [cce_mem_msg_width_lp-1:0]       mem_cmd_i
+  , input                                   mem_cmd_v_i
+  , output logic                            mem_cmd_ready_o
```
Comment (dpetrisko) [inline bp_me/src/v/axi/bp_me_cce_mem_to_axi_wrapper.v:41]: Rather than have the PISO/SIPO in here, break into two ready/valid interfaces ("BP Burst"), aligning the data width of the AXI bus and the BP bus.  Then we can move the gearboxing to in front of this module

File: bp_me/src/v/axi/bp_me_outgoing_io_to_axi_wrapper.v [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+  )
+
+  ( 
+  //==================== GLOBAL SIGNALS =======================
+    input aclk_i  
+  , input aresetn_i
+
+  //========================= BP side =========================
+  // io commands to the axi wrapper
+  , input  [cce_mem_msg_width_lp-1:0]       io_cmd_i
```
Comment (dpetrisko) [inline bp_me/src/v/axi/bp_me_outgoing_io_to_axi_wrapper.v:40]: same bp_burst comment as above.  Also rename to mem instead of I/O

File: bp_me/src/v/axi/bp_me_outgoing_io_to_axi_wrapper.v [hunk 5]
```diff
@@ -0,0 +1,393 @@
...
+
+        // else if the response is OKAY and valid
+        else if (axi_bresp_i == '0 & axi_bvalid_i) begin
+          state_n                 = e_wait;
+          io_resp_v_o             = 1'b1;
+          axi_write_piso_reset_li = 1'b1;        
+        end
+      end
+
+      e_write_err : begin
```
Comment (dpetrisko) [inline bp_me/src/v/axi/bp_me_outgoing_io_to_axi_wrapper.v:337]: I'm not familiar, what's an error condition on an AXI bus? Since we don't have a way to signal error condition in BP, I think the appropriate thing is to $error("") and return 0 to the BP system

File: bp_me/test/tb/bp_axi/bp_nonsynth_mem_to_axi_wrapper_tb.v [hunk 6]
```diff
@@ -0,0 +1,271 @@
+`include "bsg_defines.v"
+
+module bp_nonsynth_mem_to_axi_wrapper_tb
```
Comment (dpetrisko) [inline bp_me/test/tb/bp_axi/bp_nonsynth_mem_to_axi_wrapper_tb.v:3]: This seems generic enough to live in basejump_stl? Maybe make a separate PR there?

File: bp_me/test/tb/bp_axi/bp_nonsynth_mem_to_axi_wrapper_tb.v [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+
+ #(// BP PARAMS
+  parameter cce_block_width_p  = 512
```
Comment (dpetrisko) [inline bp_me/test/tb/bp_axi/bp_nonsynth_mem_to_axi_wrapper_tb.v:6]: Unused?

**Comment by dpetrisko** [comment]
@tanglingshu can we get an updated version of this PR?

**Comment by tanglingshu** [comment]
Replaced with PR #684 because I cannot correctly update my top_dev branch
